File: frontend/src/types/task.ts
export interface Task {
  id: number
  name: string
  status: 'uploading' | 'pending' | 'processing' | 'completed' | 'error'
  stage: 'initial' | 'unpacking' | 'de_duplication' | 'preview_generation' | 'focus_detection' | 'cropping' | 'tagging' | 'packaging' | 'finished'
  progress: number
  message: string
  progress_detail: {
    overall_percent: number
    stage_index: number
    stage_total: number
    stage_name: string
    stage_percent: number
    step_hint?: string
  }
  created_at: string
  updated_at: string
  focus_model: string
  tag_model: string
  stats: {
    total_files: number
    image_files: number
    kept_files: number
    processed_files: number
  }
  upload_path?: string
  export_path?: string
  export_ready?: boolean
  items?: TaskImage[]
  config?: Record<string, any>
}

export interface TaskImage {
  id: number
  task_id: number
  orig_name: string
  orig_path?: string | null
  preview_path?: string | null
  crop_path?: string | null
  prompt_txt_path?: string | null
  md5?: string
  phash?: string
  sharpness?: number
  width?: number
  height?: number
  selected: boolean
  meta_json: any
  preview_url?: string | null
  crop_url?: string | null
  has_prompt: boolean
  prompt_text?: string
  subject_area_ratio?: number
  has_face?: boolean
  face_conf?: number
  has_pose?: boolean
  pose_conf?: number
  shot_type?: string
  confidence?: number
  reason?: string
  cluster_id?: string | number
  quality?: {
    usable?: boolean
    reject_reason?: string
  }
  crop_square_model?: any
  crop_square_user?: any
  decision?: {
    keep?: boolean
  }
}

export interface TaskCreate {
  focus_model?: string
  tag_model?: string
}

export interface TaskBatchResponse {
  id: number
  zip_name: string
}

File: backend/app/schemas/task.py
from pydantic import BaseModel
from typing import Optional, Dict
from datetime import datetime
from app.models.task import TaskStatus, TaskStage


class TaskBatchResponse(BaseModel):
    id: int
    zip_name: str


class ProgressInfo(BaseModel):
    overall_percent: float
    stage_index: int
    stage_total: int
    stage_name: str
    stage_percent: float
    step_hint: Optional[str] = None


class TaskResponse(BaseModel):
    id: int
    name: str
    status: TaskStatus
    stage: TaskStage
    progress: int
    message: str
    created_at: datetime
    updated_at: datetime
    focus_model: str
    tag_model: str
    stats: Dict
    upload_path: Optional[str] = None
    export_path: Optional[str] = None
    export_ready: bool = False
    config: Dict = {}
    progress_detail: ProgressInfo
    items: Optional[list] = None

    class Config:
        from_attributes = True
